### PR TITLE
Add real readiness logic

### PR DESCRIPTION
**Pull request by howardjohn**
We can put this in more places as well (real XDS server), but wanted a minimal one that gets our test running properly

Part of https://github.com/istio/ztunnel/issues/156

### CODE REVIEW

File: src/workload.rs [hunk 1]
```diff
@@ -308,11 +311,14 @@ impl LocalClient {
         // Currently, we just load the file once. In the future, we could dynamically reload.
         let data = tokio::fs::read_to_string(self.path).await?;
         let r: Vec<Workload> = serde_yaml::from_str(&data)?;
-        let mut wli = self.workloads.lock().unwrap();
-        for wl in r {
-            info!("inserting local workloads {wl}");
-            wli.insert(wl);
+        {
```
Comment (zhlsunshine) [inline src/workload.rs:314]: want to know what's the purpose to new a block for workload insert for LocalClient?

Comment (howardjohn) [reply src/workload.rs:314]: To drop the mutex

Comment (howardjohn) [reply src/workload.rs:314]: I guess it might not be needed anymore, in initial iteration it was

File: src/app.rs [hunk 2]
```diff
@@ -35,10 +34,13 @@ pub async fn build_with_cert(
 
     let mut tasks: Vec<JoinHandle<()>> = Vec::new();
 
-    let workload_manager = workload::WorkloadManager::new(config.clone());
+    let ready = admin::Ready::new();
+    let proxy_task = ready.register_task("proxy listeners");
+
+    let workload_manager =
+        workload::WorkloadManager::new(config.clone(), ready.register_task("workload manage"));
```
Comment (stevenctl) [inline src/app.rs:41]:
```suggestion
        workload::WorkloadManager::new(config.clone(), ready.register_task("workload manager"));
```

File: src/admin/mod.rs [hunk 3]
```diff
@@ -44,33 +39,73 @@ use crate::{config, signal};
...
+    name: String,
+}
+
+impl BlockReady {
+    pub fn subtask(&self, name: &str) -> BlockReady {
+        self.parent.register_task(name)
+    }
+}
+
+impl Drop for BlockReady {
```
Comment (stevenctl) [inline src/admin/mod.rs:87]: If pretty much every `drop` is explicit, why do we need this vs some explicit "done" method or wrapping it in something like

```
fn subtask(&self, name: &str, f: SomeFnType) {
  // add to set
  f()
  // remove
}

```

Comment (howardjohn) [reply src/admin/mod.rs:87]: The closure won't work great some other cases (in XDS its going to be deeply nested when we process the first request fully). We can remove the drop but then it makes it possible to forget about it and block readiness indefinitely

**Review by stevenctl — APPROVED** [review]

**Pull request merged**